snuba/state/__init__.py

```diff
@@ -130,24 +138,40 @@ def abtest(value: Optional[Any]) -> Optional[Any]:
     return value
 
 
-def set_config(key: str, value: Optional[Any], user: Optional[str] = None) -> None:
+def set_config(
+    key: str, value: Optional[Any], user: Optional[str] = None, force: bool = False
+) -> None:
     if value is not None:
-        value = "{}".format(value).encode("utf-8")
+        enc_value = "{}".format(value).encode("utf-8")
+    else:
+        enc_value = b""
 
     try:
-        original_value = rds.hget(config_hash, key)
-        if value == original_value:
-            return
+        enc_original_value = rds.hget(config_hash, key)
+        if enc_original_value is not None and value is not None:
+            original_value = numeric(enc_original_value.decode("utf-8"))
```

Comment:
Do we need to cast to number? How about just enforcing the new value is the same as the original raw one? My motivation for asking is that I personally think it would be better to do away with `numeric` completely and just return whatever type the user actually asked for. What do you think?

Reply:
`numeric` name is a bit misleading. It basically converts a string value to the proper typed value. So in essence, it is comparing that the 2 values are of the same type.

We shouldn't get rid of `numeric` just yet. In redis, we are only storing the encoded values. There is no type information in redis. All type information is inferred. Until we get to a point where we start storing types in redis as well, it needs to stay